src/routes/Counter/index.js

```diff
@@ -2,7 +2,7 @@ import { injectReducer } from '../../store/reducers'
 
 export default (store) => ({
   path: 'counter',
-  getComponents (location, next) {
+  getComponent (nextState, next) {
```

Comment:
per latest release:
https://github.com/reactjs/react-router/blob/master/upgrade-guides/v2.2.0.md

looks like the roadmap has a strong emphasis on improving async routing, so this is def a step in the right direction